src/app/user/createtemplate/page.tsx

```diff
@@ -0,0 +1,277 @@
+    } else {
+      setSearchResults([]);
+    }
+  }, [searchInput]);
+
+  useEffect(() => {
+    const savedUnit = localStorage.getItem("weightUnit");
+    if (savedUnit) {
+      setUnit(savedUnit);
+    }
+  }, []);
+
+  // ------------------ Add, Delete, Update Exercises ----------------------------------
+
+  const handleSelectExercise = (exercise: Exercise): ExerciseActivity => {
+    const newExercise: ExerciseActivity = {
+      exerciseName: exercise.label,
+      sets: [{ setNumber: 1, reps: 0, weight: 0, unit }],
+    };
+
+    setSelectedExercises((prev) => [...prev, newExercise]);
+    setSearchInput("");
+    setSearchResults([]);
+
+    return newExercise;
+  };
+
+  const handleDeleteExercise = (index: number) => {
+    const newSelectedExercises = [...selectedExercises];
+    newSelectedExercises.splice(index, 1);
+    setSelectedExercises(newSelectedExercises);
+  };
+
+  // ---------------------  Add, Delete, Update Sets ------------------------------------
+
+  const handleAddSet = (index: number) => {
+    const newSelectedExercises = [...selectedExercises];
+    newSelectedExercises[index].sets.push({
+      setNumber: 1,
+      reps: 0,
+      weight: 0,
+      unit,
+    });
+    setSelectedExercises(newSelectedExercises);
+  };
+
+  const handleDeleteSet = (index: number, setIndex: number) => {
+    const newSelectedExercises = [...selectedExercises];
+    newSelectedExercises[index].sets.splice(setIndex, 1);
+    setSelectedExercises(newSelectedExercises);
+  };
+
+  // ----------------------------- Toggle Units ----------------------------------------
+
+  const toggleUnit = () => {
+    const newUnit = unit === "lbs" ? "kg" : "lbs";
+    const convertedExercises = selectedExercises.map((exercise) => ({
+      ...exercise,
+      sets: exercise.sets.map((set) => ({
+        ...set,
+        weight: parseInt(convertWeight(set.weight, newUnit).toFixed()),
+      })),
+    }));
+
+    setSelectedExercises(convertedExercises);
+    setUnit(newUnit);
+
+    localStorage.setItem("weightUnit", newUnit);
+  };
+
+  // -------------------------- Save Log/Template ---------------------------------------
+
+  const handleSaveLog = async () => {
+    if (!templateName) {
+      setErrorMessage("Template name can't be empty.");
+      return;
+    }
+
+    if (session?.user?._id) {
+      const logData = [
+        {
+          name: templateName,
+          exercises: selectedExercises.map((exerciseActivity) => {
+            return {
+              exerciseName: exerciseActivity.exerciseName,
+              sets: exerciseActivity.sets.map((set, index) => {
+                return {
+                  setNumber: index + 1,
+                  weight: set.weight,
+                  unit: unit,
+                  reps: set.reps,
+                };
+              }),
+            };
+          }),
+          isTemplate: true,
+        },
+      ];
+
+      await LoggingClient.saveLog({
+        userId: session.user._id,
```

Comment:
I removed the `userId` parameter from this function. We don't need to send this to the server as the server knows all about the auth session :)